django/db/backends/oracle/operations.py

```diff
@@ -310,17 +312,88 @@ def savepoint_create_sql(self, sid):
     def savepoint_rollback_sql(self, sid):
         return convert_unicode("ROLLBACK TO SAVEPOINT " + self.quote_name(sid))
 
+    @cached_property
```

Comment:
A cached property on the Operations class becomes problematic when migrations enter the scene (and there are tests with migrations, so the interaction is likely to exist already with the first use-case).